### PR TITLE
[MM-15305] Migrate Preference.CleanupFlagsBatch to Sync by default

### PR DESCRIPTION
**Pull request by farhadab**
#### Summary
This PR migrates the CleanupFlagsBatch method from the Preference store to Sync by
default.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/10712

### CODE REVIEW

**Comment by hanzei** [comment]
@jespino The tests fail because this breaks the EE code. Is there a way to run the TE tests anyway?

File: store/sqlstore/preference_store.go [hunk 1]
```diff
@@ -282,45 +282,42 @@ func (s SqlPreferenceStore) DeleteCategoryAndName(category string, name string)
...
+					LIMIT
+						:Limit
+				)
+				AS t
+			)`
+
+	sqlResult, err := s.GetMaster().Exec(query, map[string]interface{}{"Category": model.PREFERENCE_CATEGORY_FLAGGED_POST, "Limit": limit})
+	if err != nil {
+		return int64(0), model.NewAppError("SqlPostStore.CleanupFlagsBatch", "store.sql_preference.cleanup_flags_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
+	} else {
```
Comment (hanzei) [inline store/sqlstore/preference_store.go]: You can drop the `else` here.

File: store/sqlstore/preference_store.go [hunk 2]
```diff
@@ -282,45 +282,42 @@ func (s SqlPreferenceStore) DeleteCategoryAndName(category string, name string)
...
+		rowsAffected, err1 := sqlResult.RowsAffected()
```
Comment (hanzei) [inline store/sqlstore/preference_store.go]: Do you need a different name here? I don't think so.

File: store/sqlstore/preference_store.go [hunk 3]
```diff
@@ -282,45 +282,42 @@ func (s SqlPreferenceStore) DeleteCategoryAndName(category string, name string)
...
+		if err1 != nil {
+			return int64(0), model.NewAppError("SqlPostStore.CleanupFlagsBatch", "store.sql_preference.cleanup_flags_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
 		} else {
```
Comment (hanzei) [inline store/sqlstore/preference_store.go]: Here you can also drop the `else`.

Comment (farhadab) [reply store/sqlstore/preference_store.go]: Hi @hanzei , glad to help and thanks for bearing with me - it's my first contribution to open source (even though it was mentioned that this is for enterprise).

For the naming, you mean err1 can just be err, right? All that stuff I just kept as it was with the previous commit, but yes, I'll change it - I assume the following is fine?
```go

	sqlResult, err := s.GetMaster().Exec(query, map[string]interface{}{"Category": model.PREFERENCE_CATEGORY_FLAGGED_POST, "Limit": limit})
	if err != nil {
		return int64(0), model.NewAppError("SqlPostStore.CleanupFlagsBatch", "store.sql_preference.cleanup_flags_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
	}

	rowsAffected, err := sqlResult.RowsAffected()
	if err != nil {
		return int64(0), model.NewAppError("SqlPostStore.CleanupFlagsBatch", "store.sql_preference.cleanup_flags_batch.app_error", nil, ""+err.Error(), http.StatusInternalServerError)
	}

	return rowsAffected, nil
```

Comment (hanzei) [reply store/sqlstore/preference_store.go]: That looks pretty good! :+1:

**Comment by jespino** [comment]
@farhadab Now the entepreprise repo changes are approved, do you mind to merge master into the branch?

**Pull request merged**